src/main/java/core/basesyntax/Plane.java

```diff
@@ -1,11 +1,74 @@
 package core.basesyntax;
 
-/**
- * <p>Дано класс Plane. Написать класс PlaneBuilder, который реализует паттерн Builder.
- * Добавьте не менее пяти полей в класс Plain. Внести все необходимые изминения в класс Plane
- * которые могут потребоваться для реализации паттерна Builder.</p>
- */
-
 public class Plane {
+    private final String name;
+    private final String type;
+    private final int maxSpeed;
+    private final int passengerCapacity;
+    private final int rangeOfFlight;
+
+    private Plane(PlaneBuilder builder) {
+        this.name = builder.name;
+        this.type = builder.type;
+        this.maxSpeed = builder.maxSpeed;
+        this.passengerCapacity = builder.passengerCapacity;
+        this.rangeOfFlight = builder.rangeOfFlight;
```

Comment:
do we need to explicitly use `this` keyword in this case?

Reply:
I think yes, because names of variables are the same. I know that it will work without "this" but it makes code more clear.
And today Sofia posted links to two articles and in both is said that using "this" in constructors is a good practice.

Reply:
in this particular case, there is already difference between `name` and `builder.name`
and the compiler will understand your intentions if you do not write `this` keyword

but if case of the following code:
```
private String name;

public Plane(String name) {
       this.name = name;
}
```

- we need to write `this` for the compiler to understand where we are addressing the local variable passed to the constructor, and where the class field